ffi/derive/src/attr_parse/getset.rs

```diff
@@ -0,0 +1,830 @@
+fn insert_gen_request(
+    accumulator: &mut darling::error::Accumulator,
+    gen_map: &mut RequestedAccessors,
+    span: Span,
+    mode: GetSetGenMode,
+    options: GetSetOptions,
+) {
+    if options.with_prefix && mode == GetSetGenMode::Set {
+        accumulator.push(
+            darling::Error::custom("`with_prefix` is not supported for `set`").with_span(&span),
+        );
+    }
+
+    match gen_map.entry(mode) {
+        Entry::Occupied(_) => accumulator.push(
+            darling::Error::custom(format!("duplicate `getset({})` attribute", mode))
+                .with_span(&span),
+        ),
+        Entry::Vacant(v) => {
+            v.insert(options);
+        }
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Clone)]
+pub struct GetSetFieldAttr {
+    pub skip: bool,
+    pub gen: RequestedAccessors,
+}
+
+impl darling::FromAttributes for GetSetFieldAttr {
+    fn from_attributes(attrs: &[Attribute]) -> darling::Result<Self> {
+        let mut accumulator = darling::error::Accumulator::default();
+        let mut skip_span = None;
+        let mut result = GetSetFieldAttr {
+            skip: false,
+            gen: FxHashMap::default(),
+        };
+        for attr in attrs {
+            // getset crate is quite liberal in what it accepts
+            // it allows both the `#[getset(get)]` and `#[get]` syntax to be used
+            // iroha doesn't use the latter form, so it is not supported by `iroha_ffi_derive`
+            if attr.path().is_ident("getset") {
+                let Some(list) = accumulator.handle(attr.meta.require_list().map_err(Into::into))
+                    else { continue };
+                let Some(tokens): Option<Punctuated<SpannedGetSetAttrToken, Token![,]>>
+                    = accumulator.handle(list.parse_args_with(Punctuated::parse_terminated).map_err(Into::into))
+                    else { continue };
+
+                for token in tokens {
+                    match token.token {
+                        GetSetAttrToken::Skip => {
+                            result.skip = true;
+                            skip_span = Some(token.span);
+                        }
+                        GetSetAttrToken::Gen(mode, options) => insert_gen_request(
+                            &mut accumulator,
+                            &mut result.gen,
+                            token.span,
+                            mode,
+                            options,
+                        ),
+                    }
+                }
+            } else if attr
+                .path()
+                .get_ident()
+                .and_then(|ident| GetSetGenMode::from_str(&ident.to_string()).ok())
+                .is_some()
+            {
+                accumulator.push(
+                    darling::Error::custom(
+                        "getset attributes without `getset` prefix are not supported by iroha_ffi_derive",
```

Comment:
This error is repeated further.

Reply:
Yeah, the parsing these two kinds of attrs is kinda copy-pasty.. I'll see what I can do